catalog/cataloger_create_entries.go

```diff
@@ -33,9 +36,30 @@ func (c *cataloger) CreateEntries(ctx context.Context, repository, branch string
 		if err != nil {
 			return nil, err
 		}
-		for i := range entries {
-			if _, err := insertEntry(tx, branchID, &entries[i]); err != nil {
-				return nil, fmt.Errorf("entry at %d: %w", i, err)
+		// single insert per batch
+		for i := 0; i < len(entries); i += c.CreateEntriesInsertSize {
```

Comment:
When there are duplicate create-entries, this behaves unpredictably:
- If a dupe appears inside a batch, we receive an error from INSERT (and I assume a strange error message).
- If a dupe appears between batches, the latest version wins.

I don't particularly care whether it should succeed or fail, but if I had to rank them I'd go with:
1. The behaviour on AWS S3.
2. Latest version wins, always.
3. A decipherable error, always, and _no_ file is created.
4. A decipherable error, always, and _some_ of the files may be created.

In order to do any of these we _will_ have to examine all entries in the batch.

Reply:
Going with "Latest version wins"